pypeit/scripts/show_2dspec.py

```diff
@@ -127,77 +131,153 @@ def main(args):
         else:
             detname = DetectorContainer.get_name(det)
 
-        # Load it up -- NOTE WE ALLOW *OLD* VERSIONS TO GO FORTH
-        spec2DObj = spec2dobj.Spec2DObj.from_file(args.file, detname, chk_version=False)
-        # Use the appropriate class to get the "detector" number
-        det = spec2DObj.detector.parse_name(detname)
-
-        # Setup for PypeIt imports
-        msgs.reset(verbosity=args.verbosity)
-
         # Find the set of channels to show
-        if args.channels is not None:
-            show_channels = [int(item) for item in args.channels.split(',')]
-        else:
-            show_channels = [0,1,2,3]
+        show_channels = [0,1,2,3] if args.channels is None \
+                            else [int(item) for item in args.channels.split(',')]
 
-        # Grab the slit edges
-        slits = spec2DObj.slits
-        if spec2DObj.sci_spat_flexure is not None:
-            msgs.info("Offseting slits by {}".format(spec2DObj.sci_spat_flexure))
-        all_left, all_right, mask = slits.select_edges(flexure=spec2DObj.sci_spat_flexure)
-
-        # TODO -- This may be too restrictive, i.e. ignore BADFLTCALIB??
-        gpm = mask == 0
+        # Try to read the Spec2DObj using the current datamodel, but allowing
+        # for the datamodel version to be different
+        try:
+            spec2DObj = spec2dobj.Spec2DObj.from_file(args.file, detname, chk_version=False)
+        except PypeItError:
+            try:
+                # Try to get the pypeit version used to write this file
+                file_pypeit_version = fits.getval(args.file, 'VERSPYP', 0)
+            except KeyError:
+                file_pypeit_version = '*unknown*'
+            msgs.warn(f'Your installed version of PypeIt ({__version__}) cannot be used to parse '
+                      f'{args.file}, which was reduced using version {file_pypeit_version}.  You '
+                      'are strongly encouraged to re-reduce your data using this (or, better yet, '
+                      'the most recent) version of PypeIt.  Script will try to parse only the '
+                      'relevant bits from the spec2d file and continue (possibly with more '
+                      'limited functionality).')
+            spec2DObj = None
+
+        if spec2DObj is None:
+            # Try to get the relevant elements directly from the fits file
+            with io.fits_open(args.file) as hdu:
+                names = [h.name for h in hdu]
+                has_det = any([detname in n for n in names])
+                if not has_det:
+                    msgs.error(f'Provided file has no extensions including {detname}.')
+                for ext in ['SCIIMG', 'SKYMODEL', 'OBJMODEL', 'IVARMODEL']:
+                    _ext = f'{detname}-{ext}'
+                    if _ext not in names:
+                        msgs.error(f'{args.file} missing extension {_ext}.')
+
+                sciimg = hdu[f'{detname}-SCIIMG'].data
```

Comment:
should we put this code into the spec2DObj?
  or is that simply not feasible?

Reply:
It's feasible, but I don't think it makes sense.  I see this code as something specific to the script.  I.e., we don't want to change `Spec2DObj` (or worse `DataContainer`) more generally to allow it to distinguish between what is and is not "mission critical" for some subset of usage cases.  But this script presents a specific usage case that we're trying to accommodate.